dat/glsl/rounded_rect.frag

```diff
@@ -0,0 +1,25 @@
+#include "lib/sdf.glsl"
+uniform vec4 colour;
+uniform vec4 dimensions; // size, corners radius
+uniform float paramf; // line_width
+
+in vec2 pos;
+out vec4 colour_out;
+
+void main(void) {
+    vec2 uv = pos * 2.0 - 1.0;
+
+    float d = paramf + sdBoxRound(
+                uv * dimensions.xy / 2.0,
+                dimensions.xy / 2.0,
+                dimensions.zw);
+    colour_out = colour;
+    if (paramf > 0.) {
+        d = abs(d);
+        colour_out.a = smoothstep(
+                -paramf,
+                -paramf / 2.,
+                -d);
+    } else
+        colour_out.a = smoothstep(0, 0.5, -d);
```

Comment:
This actually doesn't compile on linux/nvidia. In glsl, int/float difference is much stricter than standard C. For most implementations, `0` will not be cast to float. So it errors because it tries to find a function `smoothstep( int, float, float )`. It's best to write floats with `0.0` (not just `0.` since some drivers don't like that).

Reply:
rip, i did it again.. my bad x'/